sdks/java/io/hadoop-input-format/src/main/java/org/apache/beam/sdk/io/hadoop/inputformat/HadoopInputFormatIO.java

```diff
@@ -0,0 +1,967 @@
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+
+/**
+ * A read transform for any data source which implements Hadoop InputFormat. For example- Cassandra,
+ * Elasticsearch, HBase, Redis, Postgres etc.
+ * <p>
+ * The Hadoop InputFormat source returns a set of {@link org.apache.beam.sdk.values.KV} key-value
+ * pairs returning a {@code PCollection<KV>}.
+ * <p>
+ * Hadoop Configuration is mandatory for reading data using Hadoop InputFormat source.
+ * <p>
+ * Hadoop {@link org.apache.hadoop.conf.Configuration Configuration} object will have to be set with
+ * following properties without fail:
+ * <ul>
+ * <li>mapreduce.job.inputformat.class : Signifies InputFormat class required to read data from
+ * source. For example, if user wants to read data from Cassandra using CqlInputFormat, then
+ * 'mapreduce.job.inputformat.class' property must be set to CqlInputFormat.class.</li>
+ * <li>key.class : Signifies output key class of InputFormat specified in
+ * 'mapreduce.job.inputformat.class'. For example, if class CqlInputFormat is specified in
+ * 'mapreduce.job.inputformat.class', then 'key.class' property must be set to Long.class.</li>
+ * <li>value.class : Signifies output value class of InputFormat specified in
+ * 'mapreduce.job.inputformat.class'. For example, if class CqlInputFormat is specified in
+ * 'mapreduce.job.inputformat.class', then 'key.class' property must be set to Row.class.</li>
+ * </ul>
+ * For example:
+ * 
+ * <pre>
+ * <tt>
+ *   Configuration myHadoopConfiguration = new Configuration(false);
+ *   // Set Hadoop InputFormat, key and value class in configuration
+ *   myHadoopConfiguration.setClass("mapreduce.job.inputformat.class", InputFormatClass, InputFormat.class);
+ *   myHadoopConfiguration.setClass("key.class", InputFormatKeyClass, Object.class);
+ *   myHadoopConfiguration.setClass("value.class", InputFormatValueClass, Object.class);
+ * </tt>
+ * </pre>
+ * <p>
+ * Key/value translation is a mechanism provided to user for translating InputFormat's key/value
+ * type to any other class type for which a coder should be available.
+ * <p>
+ * For example:
+ * 
+ * <pre>
+ * SimpleFunction<InputFormatKeyClass, MyKeyType> myOutputKeyType =
+ *     new SimpleFunction<InputFormatKeyClass, MyKeyType>() {
+ *       public MyKeyType apply(InputFormatKeyClass input) {
+ *         // ...logic to transform InputFormatKeyClass to MyKeyType
+ *       }
+ *     };
+ * 
+ * SimpleFunction<InputFormatValueClass, MyValueType> myOutputValueType =
+ *     new SimpleFunction<InputFormatValueClass, MyValueType>() {
+ *       public MyValueType apply(InputFormatValueClass input) {
+ *         // ...logic to transform InputFormatValueClass to MyValueType
+ *       }
+ *     };
+ * </pre>
+ * 
+ * <h3>Read configuration options supported for HadoopInputFormatIO</h3>
```

Comment:
I think we can cut this whole section (from here to the next h3), given that we're showing end to end examples of how to use it with cassandra/elasticsearch.